### PR TITLE
fix: Fix failing ddos pp pipeline runs - `avm/res/network/ddos-protection-plan`

### LINKED ISSUE #2003: [Failed pipeline] avm.res.network.ddos-protection-plan
Failed run: https://github.com/Azure/bicep-registry-modules/actions/runs/9177993060

### PR DESCRIPTION
**Pull request by segraef**
## Pipeline Reference

Closes https://github.com/Azure/bicep-registry-modules/issues/2003 (might be already closed) and to prevent future failing runs.

| Pipeline |
| -------- |
|     [avm.res.network.ddos-protection-plan](https://github.com/Azure/bicep-registry-modules/actions/workflows/avm.res.network.ddos-protection-plan.yml)     |

## Type of Change

- [ ] Update to CI Environment or utlities (Non-module effecting changes)
- [x] Azure Verified Module updates:
  - [x] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [x] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] I have run `Set-AVMModule` locally to generate the supporting module files.
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: avm/res/network/ddos-protection-plan/tests/e2e/defaults/main.test.bicep [hunk 1]
```diff
@@ -20,6 +20,9 @@ param serviceShort string = 'ndppmin'
 @description('Optional. A token to inject into the name of each resource.')
 param namePrefix string = '#_namePrefix_#'
 
+#disable-next-line no-hardcoded-location // Due to quotas and capacity challenges, this region must be used in the AVM testing subscription
```
Comment (eriqua) [inline avm/res/network/ddos-protection-plan/tests/e2e/defaults/main.test.bicep:23]: It's a very minor thing please bare with me, just in alignment with other implementations
```suggestion
// Due to quotas and capacity challenges, this region must be used in the AVM testing subscription
#disable-next-line no-hardcoded-location 
```

File: avm/res/network/ddos-protection-plan/tests/e2e/max/main.test.bicep [hunk 2]
```diff
@@ -20,6 +20,9 @@ param serviceShort string = 'ndppmax'
 @description('Optional. A token to inject into the name of each resource.')
 param namePrefix string = '#_namePrefix_#'
 
+#disable-next-line no-hardcoded-location // Due to quotas and capacity challenges, this region must be used in the AVM testing subscription
```
Comment (eriqua) [inline avm/res/network/ddos-protection-plan/tests/e2e/max/main.test.bicep:23]:
```suggestion
// Due to quotas and capacity challenges, this region must be used in the AVM testing subscription
#disable-next-line no-hardcoded-location 
```

File: avm/res/network/ddos-protection-plan/tests/e2e/waf-aligned/main.test.bicep [hunk 3]
```diff
@@ -20,6 +20,9 @@ param serviceShort string = 'ndppwaf'
 @description('Optional. A token to inject into the name of each resource.')
 param namePrefix string = '#_namePrefix_#'
 
+#disable-next-line no-hardcoded-location // Due to quotas and capacity challenges, this region must be used in the AVM testing subscription
```
Comment (eriqua) [inline avm/res/network/ddos-protection-plan/tests/e2e/waf-aligned/main.test.bicep:23]:
```suggestion
// Due to quotas and capacity challenges, this region must be used in the AVM testing subscription
#disable-next-line no-hardcoded-location 
```

**Comment by segraef** [comment]
retriggered workflow as well

**Review by ChrisSidebotham — APPROVED** [review]
🚀

**Review by fbinotto — COMMENTED** [review]
wrong page. can be deleted.

**Pull request merged**